src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContractResolver.cs

```diff
@@ -8,7 +8,7 @@ namespace System.Runtime.Serialization
 {
     public abstract class DataContractResolver
     {
-        public abstract bool TryResolveType(Type type, Type declaredType, DataContractResolver knownTypeResolver, out XmlDictionaryString typeName, out XmlDictionaryString typeNamespace);
-        public abstract Type ResolveName(string typeName, string typeNamespace, Type declaredType, DataContractResolver knownTypeResolver);
+        public abstract bool TryResolveType(Type type, Type? declaredType, DataContractResolver knownTypeResolver, out XmlDictionaryString? typeName, out XmlDictionaryString? typeNamespace);
```

Comment:
```suggestion
        public abstract bool TryResolveType(Type? type, Type? declaredType, DataContractResolver? knownTypeResolver, out XmlDictionaryString? typeName, out XmlDictionaryString? typeNamespace);
```

Reply:
Similar logic as above - we never pass `null` for `knownTypeResolver`, and so implementations of this shouldn't need to check it for null if they want to use it.